### PR TITLE
BaSP-TG-3: Create methods and resources to Admin

### PR DESCRIPTION
**Pull request by JulianFloresDev**
Create:

-  Admin Schema
- admin routes to:
- - Get by Id
- - Delete by Id
- - Edit Admin by Id

### CODE REVIEW

File: src/controllers/admins.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    );
+
+    return res.status(202)
+      .json({
+        message: `Admin with id ${id} found and successfully edited!`,
+        data: admin,
+        error: false,
+      });
+  } catch (err) {
+    return res.status(408)
```
Comment (ignacioaristo) [inline src/controllers/admins.js:58]: lets change this code too

File: src/controllers/admins.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      });
+  }
+};
+
+const editAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const admin = await Admins.findByIdAndUpdate(
+      { _id: id },
+      { ...req.body },
```
Comment (ignacioaristo) [inline src/controllers/admins.js:47]: i think spread operator here is not necessary, we could do `req.body` and we get the same result.

File: src/controllers/admins.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    return res.status(202)
+      .json({
+        message: `Admin with id ${id} found and successfully edited!`,
+        data: admin,
+        error: false,
+      });
+  } catch (err) {
+    return res.status(408)
+      .json({
+        message: 'Something was wrong!',
```
Comment (ignacioaristo) [inline src/controllers/admins.js:60]: Same error message

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+const getAdminById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const admin = await Admins.findById(id);
+
+    return res.status(302).json({
+      message: 'Admin Found',
+      data: admin,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:11]: Is not necessary to show the error when an error occurs

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import Admins from '../models/Admins';
+
+const getAdminById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const admin = await Admins.findById(id);
+
+    return res.status(302).json({
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/controllers/admins.js:8]: Changes done!

File: src/controllers/admins.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      message: 'Admin Found',
+      data: admin,
+      error: false,
+    });
+  } catch (err) {
+    return res.status(408)
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/controllers/admins.js:14]: Change done!

File: src/controllers/admins.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      .json({
+        message: 'Something was wrong!',
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/controllers/admins.js:16]: I didn't know that was possible, I mean to manipulate the error received from the catch for the error message.

Change in progress!

File: src/controllers/admins.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    const admin = await Admins.findByIdAndDelete(id);
+
+    return res.status(200)
+      .json({
+        message: `Admin with id: ${id} was successfully deleted!`,
+        data: admin,
+        error: false,
+      });
+  } catch (err) {
+    return res.status(408)
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/controllers/admins.js:34]: Change in progress

File: src/controllers/admins.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      .json({
+        message: 'Something was wrong!',
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/controllers/admins.js:36]: Change Done

File: src/controllers/admins.js (hunk 1 above)
Comment (JulianFloresDev) [reply src/controllers/admins.js:58]: Change Done

File: src/controllers/admins.js (hunk 2 above)
Comment (JulianFloresDev) [reply src/controllers/admins.js:47]: I honestly didn't know why Capi did that.

Comment (JulianFloresDev) [reply src/controllers/admins.js:47]: Change Done

File: src/controllers/admins.js (hunk 3 above)
Comment (JulianFloresDev) [reply src/controllers/admins.js:60]: Done

File: src/controllers/admins.js (hunk 4 above)
Comment (JulianFloresDev) [reply src/controllers/admins.js:11]: I'm not sure what you mean here.

Is about 'error' property? That is not necessary?

File: src/controllers/admins.js (hunk 2 above)
Comment (JulianFloresDev) [reply src/controllers/admins.js:47]: The slint does not allow me to remove the spread operator; it tells me that it is missing a comma.

File: src/controllers/admins.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    return res.status(200).json({
+      message: 'Admin Found',
+      data: admin,
+      error: false,
+    });
+  } catch (err) {
+    return res.status(404)
+      .json({
+        message: `Something was wrong: ${err.message}`,
+        error: err,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:17]: I think now, this prop is unnecessary

File: src/controllers/admins.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+const deleteAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const admin = await Admins.findByIdAndDelete(id);
+
+    return res.status(200)
+      .json({
+        message: `Admin with id: ${id} was successfully deleted!`,
+        data: admin,
+        error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:31]: We don't need this prop too.

File: src/controllers/admins.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+      });
+  } catch (err) {
+    return res.status(404)
+      .json({
+        message: `Something was wrong: ${err.message}`,
+        error: err,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:37]: The same here.

File: src/controllers/admins.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+      { _id: id },
+      { ...req.body },
+      { new: true },
+    );
+
+    return res.status(200)
+      .json({
+        message: `Admin with id ${id} found and successfully edited!`,
+        data: admin,
+        error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:55]: And here too.

File: src/controllers/admins.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+      });
+  } catch (err) {
+    return res.status(404)
+      .json({
+        message: 'Something was wrong!',
+        error: err,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:61]: Delete this.

File: src/controllers/admins.js [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+    return res.status(200)
+      .json({
+        message: `Admin with id ${id} found and successfully edited!`,
+        data: admin,
+        error: false,
+      });
+  } catch (err) {
+    return res.status(404)
+      .json({
+        message: 'Something was wrong!',
```
Comment (ignacioaristo) [inline src/controllers/admins.js:60]: Handle the message error like before.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

**Comment by JulianFloresDev** [comment]
All suggested changes have been implemented!

**Pull request merged**